Update dev exp landing (#537)
canonical/launchpad-manual: docs/developer/explanation/index.rst

.. meta::
   :description: Documentation covering Launchpad ideas, setup, 
      framework, parts, best practices, and deployment processes.

Explanation
===========

Launchpad is a large platform built over many years on the Zope Toolkit,
PostgreSQL, and a substantial body of Python and JavaScript. These pages

> explain the ideas behind the project, the components that make up the platform,

Reviewer: I don't think we need to include technical details here.

I would change it to something like this.
> Launchpad is a large platform built over many years. These pages explain the ideas behind the project, its components, and the best practices that its developers follow.

and the practices its developers follow.

The ideas behind Launchpad
--------------------------

Launchpad connects the networks of people who make software with the network of

> dependencies between their projects. Understanding why Launchpad exists, who it

Reviewer: ```suggestion
dependencies between their projects. Understanding why Launchpad exists, whom it
```

> dependencies between their projects. Understanding why Launchpad exists, who it

Reviewer: I don't completely understand what you mean by the first sentence.

Author: Modified the sentence to make it less abstract.

serves, and the values that guide it provides the context for everything else.

- :ref:`What is Launchpad? <what-is-launchpad>`
- :ref:`About Launchpad security <about-launchpad-security>`
- :ref:`Launchpad strategy <launchpad-strategy>`
- :ref:`Launchpad values <launchpad-values>`

Core components and setup
-------------------------

Launchpad is made up of many interdependent components. Knowing how these

> pieces fit together helps you navigate the codebase and run a working instance.

Reviewer: ```suggestion
Launchpad is made up of many components. Knowing how these
pieces fit together helps you navigate the codebase and test your changes.
```

Database
--------

Launchpad stores its data in PostgreSQL and depends heavily on

> PostgreSQL-specific features such as stored procedures, triggers, and language

Reviewer: See https://github.com/canonical/launchpad-manual/pull/537/changes#r3421500439

Author: Addressed. The information is from an existing doc

Author: I will drop these from this paragraph because they are also mentioned in the other landing page.

Database
--------

Launchpad stores its data in PostgreSQL and depends on PostgreSQL-specific
features. Its schema is patched on a live system and managed with Storm, the

> ORM Launchpad uses.

Author: ```suggestion
Launchpad uses PostgreSQL and depends on PostgreSQL-specific features. 
Its schema is patched on a live system and managed by Storm, the ORM
Launchpad uses.
```

Author: Addressed offline

Developing for the Launchpad project
------------------------------------

> Contributing to Launchpad means working across several long-lived branches,

Reviewer: > several long-lived branches

I don't think this is something we want to advertise. 😅

- :ref:`XXX policy <xxx-policy>`

Best practices for development
------------------------------

Launchpad aims to be fast, always available, and safe to change. These
principles inform how its code is structured, how performance and permissions

> are handled, and how its templates and tests are written.

Reviewer: WDYM to say through "safe to change"?

Author: It was supposed to mean safe to change with a low risk of breaking other parts of the system. I'll reword it to make this clearer.

JavaScript
----------

> Launchpad's front end is written in JavaScript built on the YUI library, with

Reviewer: Same comment about "front end" or "frontend"

JavaScript
----------

Launchpad's interactive front end is written in JavaScript built on the YUI
library, with its own build system and both unit and integration test

> frameworks.

Reviewer: ```suggestion
Launchpad's frontend is written in JavaScript, built using the YUI
library, with its own build system and both unit and integration test
frameworks.
```

What are you referring to by "its own build system"?

Author: The JavaScript build system described in this doc https://documentation.ubuntu.com/launchpad/developer/explanation/javascript/javascript-buildsystem/, but I will ask Goulin to take a look because the doc mentions that the feature is in the process of being depracated.

Author: @goulinkh corrected the other UI-related paragraph. The mention of a single launchpad.js file is in accurate, even though it appears in this doc https://documentation.ubuntu.com/launchpad/developer/explanation/javascript/javascript-buildsystem/

Static assets
-------------

Launchpad's visual style comes from CSS and images that are compiled and
optimized as part of the build, including CSS sprites that combine many small

> images into one.

Reviewer: ```suggestion
optimized as part of the build using CSS sprites.
```
